packages/salesforcedx-vscode-core/src/commands/forceApexLogGet.ts

```diff
@@ -0,0 +1,205 @@
+      );
+      vscode.window.showTextDocument(document);
+    }
+  }
+}
+
+export type ApexDebugLogIdStartTime = {
+  id: string;
+  startTime: string;
+};
+
+export type ApexDebugLogObject = {
+  Id: string;
+  StartTime: string;
+  LogLength: number;
+  Operation: string;
+  Request: string;
+};
+
+interface ApexDebugLogItem extends vscode.QuickPickItem {
+  id: string;
+}
+
+export class LogFileSelector
+  implements ParametersGatherer<ApexDebugLogIdStartTime> {
+  public async gather(): Promise<
+    CancelResponse | ContinueResponse<ApexDebugLogIdStartTime>
+  > {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const logInfos = await ForceApexLogList.getLogs(cancellationTokenSource);
+    if (logInfos.length > 0) {
+      const logItems = logInfos.map(logInfo => {
+        const icon = '$(file-text) ';
+        return {
+          id: logInfo.Id,
+          label: icon + logInfo.Operation,
+          detail: new Date(logInfo.StartTime).toLocaleTimeString('en-US', {
+            year: 'numeric',
+            month: 'numeric',
+            day: 'numeric',
+            hour: '2-digit',
+            minute: '2-digit',
+            second: '2-digit'
+          }),
+          description: `${(logInfo.LogLength / 1024).toFixed(2)} KB`
+        } as ApexDebugLogItem;
+      });
+      const logItem = await vscode.window.showQuickPick(
+        logItems,
+        { placeHolder: nls.localize('force_apex_log_get_pick_log_text') },
+        cancellationTokenSource.token
+      );
+      if (logItem) {
+        return {
+          type: 'CONTINUE',
+          data: { id: logItem.id, startTime: logItem.detail! }
+        };
+      }
+    } else {
+      return {
+        type: 'CANCEL',
+        msg: nls.localize('force_apex_log_get_no_logs_text')
+      } as CancelResponse;
+    }
+    return { type: 'CANCEL' };
+  }
+}
+
+export class ForceApexLogList {
+  public static async getLogs(
+    cancellationTokenSource: vscode.CancellationTokenSource
+  ): Promise<ApexDebugLogObject[]> {
+    const execution = new CliCommandExecutor(
+      new SfdxCommandBuilder()
+        .withDescription(nls.localize('force_apex_log_list_text'))
+        .withArg('force:apex:log:list')
+        .withArg('--json')
+        .build(),
+      { cwd: vscode.workspace.workspaceFolders![0].uri.fsPath }
+    ).execute();
+    CancellableStatusBar.show(execution, cancellationTokenSource);
+    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+    notificationService.reportExecutionError(
+      execution.command.toString(),
+      (execution.processErrorSubject as any) as Observable<Error | undefined>
+    );
+    const cmdOutput = new CommandOutput();
+    const result = await cmdOutput.getCmdResult(execution);
+    try {
+      const apexDebugLogObjects: ApexDebugLogObject[] = JSON.parse(result)
+        .result;
+      apexDebugLogObjects.sort(function(a, b): number {
```

Comment:
Use the anonymous function syntax (a, b) => {}.

Also, this is what I'm worried about. What order does get log return stuff? Can we change the order? Why must we do a sort here again?

Reply:
It's ordered by starttime in chronological order. We could change the command to order by starttime in descending order instead. Should I file a bug for that?
![image](https://user-images.githubusercontent.com/4503716/36571162-3f9a6d7c-17ea-11e8-8f4c-ae43783f253f.png)

Reply:
Yeah. Please file a bug for that. For now, keep your sorting function in since we do need it in reverse chronological order.